### ISSUE TITLE
Restart required to view new Relationship in GraphQL

### ISSUE THREAD

**Issue by pke11y** [nautobot/nautobot#446]
### Environment
* Python version: 3.8.5
* Nautobot version: 1.0.1

### Steps to Reproduce
1. Create a new Relationship e.g. between dcim.device and ipam.vlan called `device_vlan`
2. Go to the GraphiQL interface
3. Query for all devices and include the relationship attribute created in step 1 e.g. 

```
{
  devices{
    rel_device_vlan {
      id
    }
  }
}
```

### Expected Behavior
The relationship attribute (e.g. `rel_device_vlan`) is available for selection the GraphIQL interface and is processed and values returned as part of the query.

### Observed Behavior
The relationship attribute (e.g. `rel_device_vlan`) is not available on the models as per definition. The nautobot service needs to be restarted for the relationship to be loaded and be available for querying.

```
{
  "errors": [
    {
      "message": "Cannot query field \"rel_device_vlan\" on type \"DeviceType\".",
      "locations": [
        {
          "line": 37,
          "column": 5
        }
      ]
    }
  ]
}
```

**Comment by jedelman8** [comment]
Would like to make this broader in scope in that we support the ability to restart Nautobot services from the UI. Will determine if that needs to be a seperate issue.

**Comment by pke11y** [comment]
Yeah makes sense. Any change for this would be quite significant.

**Comment by glennmatthews** [comment]
Probably also an issue with other "dynamic" schema content - custom fields, computed fields in particular.

**Comment by hellerve** [comment]
I worked on this for a while now, sadly none of the approaches that we could come up with proved viable. For posterity, let me document them:

- The simplest way to make this work is to generate the schema everytime it is needed (i.e. on every API call). Unfortunately, this is prohibitively slow (on the order of seconds).
- It would also be possible to only recalculate it when it changes, based on Django signals. This proved not to be viable, either, since it would break if multiple instances of Nautobot are deployed behind a load balancer; only the instance saving the model (Relationship or CustomField) would update the API and queries to the GraphQL API would become non-deterministic.
- Finally, we could cache the schema in the database as a singleton, pickled. This does not work, either, since the classes are generated on-the-fly, and pickle does not work well with that. The class definition itself is not pickled, so any instance of NautoBot other than the one that generated the Schema would not be able to unpickle the definition (furthermore, local types are generated, and pickle does not work at all with non-global types).

If there are any approaches that we haven’t tried, I’m sure we’d all be happy to hear about them. Otherwise, the unfortunate answer is that we can’t currently fix this issue.

**Comment by glennmatthews** [comment]
Note that the test case `test_query_relationship_associations` in `nautobot/core/tests/test_graphql.py` (added by #1230) is currently skipped due to this issue. When this issue is fixed, that test case should be re-enabled and validated.

**Comment by jvanderaa** [comment]
@bryanculver I assume that this will not be fixed in 2.0 and will need to continue to be a thing to be aware of?

**Comment by lampwins** [comment]
Correct
